packages/react-reconciler/src/SchedulingProfiling.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Lane, Lanes} from './ReactFiberLane';
+import type {Fiber} from './ReactInternalTypes';
+
+import {enableSchedulingProfiling} from 'shared/ReactFeatureFlags';
+import {getStackByFiberInDevAndProd} from './ReactFiberComponentStack';
+
+/**
+ * If performance exists and supports the subset of the User Timing API that we
+ * require.
+ */
+const supportsUserTiming =
+  typeof performance !== 'undefined' && typeof performance.mark === 'function';
+
+function formatLanes(laneOrLanes: Lane | Lanes): string {
+  return ((laneOrLanes: any): number).toString();
+}
+
+export function markCommitStarted(lanes: Lanes): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--commit-start-${formatLanes(lanes)}`);
+    }
+  }
+}
+
+export function markCommitStopped(): void {
+  if (enableSchedulingProfiling) {
+    if (supportsUserTiming) {
+      performance.mark(`--commit-stop`);
+    }
+  }
+}
+
+const PossiblyWeakMap = typeof WeakMap === 'function' ? WeakMap : Map;
+// $FlowFixMe: Flow cannot handle polymorphic WeakMaps
+const wakeableIDs: WeakMap<Wakeable, number> = new PossiblyWeakMap();
+let wakeableID: number = 0;
+function getWakeableID(wakeable: Wakeable): number {
+  if (!wakeableIDs.has(wakeable)) {
+    wakeableIDs.set(wakeable, wakeableID++);
+  }
+  return ((wakeableIDs.get(wakeable): any): number);
+}
+
+export function markComponentSuspended(
+  componentName: string,
+  fiber: Fiber,
```

Comment:
nit: Since we're passing the `Fiber` in question to these mark methods, we don't really need to pass the name as well. We could always call `getComponentName` in the mark methods.